ooni/tests/mocks.py

```diff
@@ -242,3 +242,20 @@ def isSupported(self):
 
     def isReachable(self):
         return defer.succeed(True)
+
+
+class MockDnsResponsePayload(object):
+    def dottedQuad(self):
+        return '127.0.0.1'
+
+
+class MockDnsResponse(object):
+    payload = MockDnsResponsePayload()
+
+
+@defer.inlineCallbacks
+def mock_dns_lookup_address():
```

Comment:
The issue you have in the PR is in this function.

The decorator `defer.inlineCallback` can only be used when the function is a generator (i.e. it should `yield` something).

In this case since you don't actually need to "wait" on any deferred and just want to return immediately the mocked value you should change this to just be a function returning an already called deferred, like so:

```
# remove the defer.inlineCallbacks decorator
def mock_dns_lookup_address():
    return defer.succeed(
       ([MockDnsResponse()], )
    )
```